services/galley/src/Galley/Options.hs

```diff
@@ -5,126 +5,58 @@
+                                             --   disables journaling)
+    -- Logging
+    , _optLogLevel      :: !Level            -- ^ Log level (Debug, Info, etc)
+    , _optLogNetStrings :: !Bool             -- ^ Use netstrings encoding:
+                                             --   <http://cr.yp.to/proto/netstrings.txt>
     }
 
 deriveFromJSON toOptionFieldName ''Opts
 makeLenses ''Opts
 
-parseOptions :: IO Opts
-parseOptions = execParser (info (helper <*> optsParser) desc)
-  where
-    desc = header "Galley - Conversation Service" <> fullDesc
-
-optsParser :: Parser Opts
-optsParser = Opts <$>
-    (Endpoint <$>
-        (textOption $
-            long "host"
-            <> value "*4"
-            <> showDefault
-            <> metavar "HOSTNAME"
-            <> help "Hostname or address to bind to")
-        <*>
-        (option auto $
-            long "port"
-            <> short 'p'
-            <> metavar "PORT"
-            <> help "Port to listen on"))
-  <*> cassandraParser
-  <*> brigParser
-  <*> gundeckParser
-  <*> optional discoUrlParser
-  <*> settingsParser
-  <*> optional journalOptsParser
-  where
-    brigParser :: Parser Endpoint
-    brigParser = Endpoint
-        <$> (textOption $
-                long "brig-host"
-                <> metavar "HOSTNAME"
-                <> help "Brig hostname")
-        <*>  (option auto $
-                long "brig-port"
-                <> metavar "PORT"
-                <> help "Brig port")
-
-    gundeckParser :: Parser Endpoint
-    gundeckParser = Endpoint
-        <$> (textOption $
-                long "gundeck-host"
-                <> metavar "HOSTNAME"
-                <> help "Gundeck hostname")
-        <*> (option auto $
-                long "gundeck-port"
-                <> metavar "PORT"
-                <> help "Gundeck port")
-
-    settingsParser :: Parser Settings
-    settingsParser = Settings
-        <$> (option auto $
-                long "http-pool-size"
-                <> metavar "SIZE"
-                <> showDefault
-                <> help "number of connections for the http client pool"
-                <> value 128)
-        <*>
-            (option auto $
-                long "team-max-size"
-                <> metavar "INT"
-                <> help "Max. # of members in a team.")
-        <*>
-            (option auto $
-                long "conv-max-size"
-                <> metavar "INT"
-                <> help "Max. # of members in a conversation.")
-        <*>
-            (switch $
-                long "intra-device-listing"
-                <> help "Use this option if you want to fetch the device list from brig instead.")
-        <*>
-            (httpsUrlOption $
-                long "conversation-code-prefix"
-                <> metavar "STRING"
-                <> help "URI prefix for conversations with access mode 'code'")
-
+-- Used by the journaler
 journalOptsParser :: Parser JournalOpts
```

Comment:
I suppose this function could move to https://github.com/wireapp/wire-server/blob/develop/services/galley/journaler/src/Options.hs instead?

Reply:
Good idea.